Fix preloader softlocks caused by bar-scaling animation bugs

kmeisthax opened the pull request:
Fixes #1214, #367 and maybe more issues involving preloader softlocks but I haven't checked the entire issue queue for it.

The underlying problem as discovered by @adrian17 is that `xscale` and `yscale` are stored as percentages in Flash Player, but unit proportions (e.g. 0 to 1) in Ruffle. This means accumulating a total in them (e.g. for animations) causes rounding errors. I changed `DisplayObjectBase` to always store scales as `Percent` (for unit bookkeeping) and then convert to unit proportions for all the matrix math display objects have to do. This doesn't appear to have any immediately obvious performance issues, and it fixes the bug.

Herschel commented on core/src/lib.rs:
```diff
@@ -28,6 +28,7 @@ mod font;
 mod html;
 mod library;
 pub mod loader;
+mod percentage;
```
Nit: let's name the module `percent` too

kmeisthax replied:
Mooted: I'm naming it `types` instead and adding a `Degrees` type.

Herschel reviewed (changes requested):
Thank you! Could you add a test for this too? Something as simple as:
```
_xscale = 12;
_xscale += 2;
trace(_xscale);
```

`_rotation` probably has similar issues. `_rotation` is in degrees but gets transformed to radians ASAP in the setter and then cached, but instead should be cached as degrees.  (If you do add a Degrees type in this PR, maybe we should put them both in a `types` module instead of having separate modules)

`_alpha` is also a percentage (although it isn't cached, so this specific issue doesn't happen there, and in fact, in some future PR, all of ColorTransform needs to eventually turn into fixed88, #193).

FYI for your own sanity, these are different in AS3 land; `scaleX`, `scaleY`, and `alpha` are all normalized 0-1 and are cached as such; scaling by 100 anywhere will result in the same kind of precision issues. So there will eventually have to be a conditional in here based on the SWF AS version. (`rotation` is still in degrees.)

kmeisthax commented:
I've added a test that checks `_xscale` and `_yscale` both - it turns out you need about 30% or so of range in order to actually trip this check, so I checked about 200 steps of 2. I also added `_rotation` and a `Degrees` type to cache these properties in their correct unit.

Thanks for the tip about AS3 - I'll make sure to write an AS3 version of this test when the time comes so that it can have it's angles and scales stored correctly, too.

Herschel commented:
Thx!
